core/src/validator.rs

```diff
@@ -171,43 +172,37 @@ impl Validator {
             bank.slots_per_segment(),
         );
         let bf = bank_forks.clone();
-        let kp = keypair.clone();
         let mut counter = 0;
         let counter_ref = &mut counter;
-        if counter == 0 {
-            set_handler(move || { 
+        set_handler(move || { 
+                if *counter_ref == 1 {
```

Comment:
it says: error[E0594]: cannot assign to `*counter_ref`, as `Fn` closures cannot mutate their captured variables . i  googled it, peopel say to use Arc::new(0) in this case. Any suggestions?

Reply:
Ah, yeah using an Arc to hold the value sounds good.  Maybe use an atomic as well (`Arc::new(AtomicUsize::new(0));`), not 100% sure if set_handler() would be re-entered if multiple Ctrl+Cs are pressed quickly